packages/neos-ui/src/Containers/Modals/SelectNodeType/index.js

```diff
@@ -182,10 +188,97 @@ export default class SelectNodeType extends PureComponent {
         );
     }
 
-    handleNodeTypeFilterChange = filterSearchTerm => this.setState({filterSearchTerm});
+    updateFilteredNodes() {
+        const {i18nRegistry} = this.props;
+        const {filterSearchTerm} = this.state;
+
+        const allowedNodeTypes = this.getAllowedNodeTypesByCurrentInsertMode();
+
+        const filteredNodeTypesFlat = [];
+        const filteredNodeTypesGrouped = [];
+        allowedNodeTypes.map(value => {
+            const filteredNodeTypes = (value.nodeTypes || [])
+                .filter(nodeType => {
+                    if (!filterSearchTerm || filterSearchTerm === '') {
+                        return true;
+                    }
+                    const label = i18nRegistry.translate(nodeType.label, nodeType.label);
+                    if (label.toLowerCase().search(escaperegexp(filterSearchTerm.toLowerCase())) !== -1) {
+                        return true;
+                    }
+                    return false;
+                });
+
+            if (filteredNodeTypes.length > 0) {
+                filteredNodeTypesFlat.push(...filteredNodeTypes);
+                filteredNodeTypesGrouped.push(value);
+            }
+
+            return true;
+        });
+
+        const focusedNodeType = (filterSearchTerm !== '' && filteredNodeTypesFlat.length > 0) ? filteredNodeTypesFlat[0].name : null;
```

Comment:
It happens that the property name is not defined. that leads to errors in the console.
```Uncaught TypeError: can't access property "name", filteredNodeTypesFlat[(index + 1)] is undefined```

Reply:
that was not the root case but added a check to be save that it is working